sql/hive/src/main/scala/org/apache/spark/sql/hive/MetastoreRelation.scala

```diff
@@ -141,7 +142,16 @@ private[hive] case class MetastoreRelation(
           sparkSession.sessionState.conf.defaultSizeInBytes
         })
     }
-  )
+    val tableParams = hiveQlTable.getParameters
+    val rowCount = tableParams.get(AnalyzeTableCommand.ROW_COUNT)
+    if (rowCount != null && rowCount.toLong >=0) {
```

Comment:
`Statistics(sizeInBytes, Some(sizeInBytes), Option(rowCount).map(_.toLong))` is shorter than this if-else block.